pkg/api/types.go

```diff
@@ -478,6 +480,20 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// MetadataVolumeSource represents an host directory mounted on a POD. It stores pod's metadata info
+type MetadataVolumeSource struct {
+	// MetadataFiles is a list of metadata file name
+	MetadataFiles []MetadataFile `json:"metadataFile",omitempty"`
+}
+
+// MetadataFile represents and
+type MetadataFile struct {
+	// Required: this must be a C_IDENTIFIER
```

Comment:
Why does it have to be a C_IDENTIFIER?  That's more restrictive than Secrets, for example.  Why can't this be an arbitrary string, exclusive of \0 or / ?  Heck, why not allow slashes and let users define nesting?